backend/app/core/background_runner.py

```diff
@@ -0,0 +1,54 @@
+import asyncio
+import json
+
+from app.core.bot import BaseBotClient
+from app.core.config import settings
+
+
+class BackgroundRunner:
+    def __init__(self):
+        self.client = BaseBotClient.get_bot_client()
+        self.background_task = None
+
+    async def start_bot(self):
+        try:
+            with open(settings.matrix_bot.next_batch_token_file, "r") as f:
+                data = json.load(f)
+                self.client.next_batch = data["next_batch_token"]
+        except IOError:
+            print(f"Couldn't load next batch token from file.")
+        except json.JSONDecodeError:
+                print("Couldn't read JSON file; overwriting")
+
+        try:
+            await self.client.login()
+            await self.client.sync_forever(
+                30000,
+                since=self.client.next_batch,
+                full_state=True,
+                loop_sleep_time=2000,
+            )
+            await asyncio.sleep(1)
```

Comment:
Why do we need this?

Reply:
It's actually a weird behaviour that I have noticed. Without that `asyncio.sleep(1)` the client is not closing on app shutdown.

Even `asyncio.sleep(0)` triggers the same error.

Reply:
Can you add a comment above this line explaining it? We should probably try and understand this properly at some point.